ios-training-bjung/Models/WeatherModel.swift

```diff
@@ -23,4 +24,8 @@ final class WeatherModel: WeatherModelInput {
     func fetchWeatherData() async throws -> String {
         return yumemiWeather.fetchWeatherCondition()
     }
+    
+    func fetchWeatherData(location: String) async throws -> String {
```

Comment:
Swift API Design Guidelines では「一般的な API は英語のフレーズを意識して表現し、前置詞句で表現すべきところでは通常、その前置詞句を引数ラベルに使用する」というものがあります。

今回の API では「// fetch weather data at `location`.」と表現すると自然な表記になると思うので、そのようなときには、次の API 宣言が Swift らしく感じられるかもしれません。

```suggestion
    func fetchWeatherData(at location: String) async throws -> String {
```

[Swift API Design Guidelines で推奨されるラベル表記方法](https://www.swift.org/documentation/api-design-guidelines/#argument-labels) において、`location` を取るべきところで引数ラベルに `location` と記載することはあまりなく、それが推奨されるのはたとえば「イニシャライザーの引数」が特例として挙げられていたり、文法的なフレーズの中にその location が含まれないときとして挙げられています。

または、今回のように `at` ラベルで `String` 型の値を受け取る場合、それが「何の文字列か」汲み取りにくくなると判断したときには、その意味を補足するために、ラベル名を `atLocation` としたりすることも [ガイドラインで推奨](https://www.swift.org/documentation/api-design-guidelines/#argument-labels:~:text=Compensate%20for%20weak%20type%20information%20to%20clarify%20a%20parameter%E2%80%99s%20role.) されています。

```swift
    func fetchWeatherData(atLocation location: String) async throws -> String {
```

Reply:
@es-kumagai 

ご指摘ありがとうございます。なるほど・・・
2f0448f6a764d025878260ab469512ad2adabc9a
上記のコミットで外部引数名を at で使うようにしました。